pkg/occlient/occlient.go

```diff
@@ -1056,32 +1056,50 @@ func (c *Client) FollowBuildLog(buildName string) error {
 // kind is always required (can be set to 'all')
 // name can be omitted if labels are set, in that case set name to ''
 // if you want to delete object just by its name set labels to nil
-func (c *Client) Delete(kind string, name string, labels map[string]string) (string, error) {
+func (c *Client) Delete(labels map[string]string) error {
+	// convert labels to selector
+	selector := util.ConvertLabelsToSelector(labels)
 
-	args := []string{
-		"delete",
-		kind,
+	// Delete DeploymentConfig
+	err := c.appsClient.DeploymentConfigs(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return errors.Wrap(err, "unable to delete deploymentconfig")
 	}
-
-	if len(name) > 0 {
-		args = append(args, name)
+	// Delete Route
+	err = c.routeClient.Routes(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return errors.Wrap(err, "unable to delete route")
 	}
-
-	if labels != nil {
-		var labelsString []string
-		for key, value := range labels {
-			labelsString = append(labelsString, fmt.Sprintf("%s=%s", key, value))
+	// Delete BuildConfig
+	err = c.buildClient.BuildConfigs(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return errors.Wrap(err, "unable to delete buildconfig")
+	}
+	// Delete Build
+	err = c.buildClient.Builds(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return errors.Wrap(err, "unable to delete build")
+	}
+	// Delete ImageStream
+	err = c.imageClient.ImageStreams(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return errors.Wrap(err, "unable to delete imagestream")
+	}
+	// Delete Services
+	svcList, err := c.kubeClient.CoreV1().Services(c.namespace).List(metav1.ListOptions{LabelSelector: selector})
```

Comment:
Do we know why it is not possible to use `DeleteCollection` for Services?

Reply:
@kadel , no :(